### DISCUSSION TITLE
Attempted import error: 'experimental_useEffectEvent' is not exported from 'react'

### DISCUSSION THREAD

**Discussion by vegandiet705** [vercel/next.js#63650] [Help]
### Summary

I got this error when running `next build` on my project. I'm trying to use the experimental React hook `useEffectEvent`. Anyone had success using this hook with Next.js ?

Thanks in advance

**Comment by icyJoseph** [comment]
Hi,

This API is not even present in canary AFAIK. You have you use experimental as per docs:

- https://react.dev/reference/react/experimental_useEffectEvent

**Comment by JeffBerger** [comment]
I haven't been able to import and use the useEffectEvent from react@experimental in my NextJS app either despite following said instructions.. It'd really simplify a lot of code if I could.

**Reply by icyJoseph** [reply]
Can you try to enable taint? That opts in your app into react@experimental iirc

- https://nextjs.org/docs/app/api-reference/config/next-config-js/taint

**Reply by ulysses-ck** [reply]
> I haven't been able to import and use the useEffectEvent from react@experimental in my NextJS app either despite following said instructions.. It'd really simplify a lot of code if I could.

I'm trying to use the useKeyPress hook that needs the experimental useEffectEvent, but I'm getting "react__WEBPACK_IMPORTED_MODULE_0__.useEffectEvent is not a function". Did you know if I need to do an extra step?

https://usehooks.com/usekeypress

**Comment by m3h3d1ha2an** [comment]
## Error Type
Runtime TypeError

## Error Message
(0 , __TURBOPACK__imported__module__$5b$project$5d2f$node_modules$2f2e$pnpm$2f$next$40$15$2e$5$2e$6_react$2d$dom$40$19$2e$2$2e$0_react$40$19$2e$2$2e$0_$5f$react$40$19$2e$2$2e$0$2f$node_modules$2f$next$2f$dist$2f$compiled$2f$react$2f$index$2e$js__$5b$app$2d$client$5d$__$28$ecmascript$29$__.useEffectEvent) is not a function


    at ModePageWrapper (src/app/dashboard/users/[mode]/mode-page-wrapper.tsx:12:46)
    at UserFormPage (src\app\dashboard\users\[mode]\page.tsx:37:5)

## Code Frame
  10 |   const router = useRouter()
  11 |   const canAccess = useHasPermission("Users", mode)
> 12 |    const redirectToDashboard = useEffectEvent(() => {
     |                                              ^
  13 |     router.push("/dashboard")
  14 |   })
  15 |

Next.js version: 15.5.6 (Turbopack)

**Reply by icyJoseph** [reply]
Try with next@canary, that'll include these. Don't use canary in production though.
